src/Microsoft.AspNet.Mvc.Localization/HtmlLocalizer.cs

```diff
@@ -0,0 +1,246 @@
+            set
+            {
+                _localizer = value;
+            }
+        }
+
+        private IStringLocalizerFactory LocalizerFactory
+        {
+            get
+            {
+                if (_localizerFactory == null)
+                {
+                    throw new InvalidOperationException(Resources.NullStringLocalizerFactory);
+                }
+
+                return _localizerFactory;
+            }
+            [param: NotNull]
+            set
+            {
+                _localizerFactory = value;
+            }
+        }
+
+        /// <inheritdoc />
+        public virtual LocalizedString this[[NotNull] string key] => _localizer[key];
+
+        /// <inheritdoc />
+        public virtual LocalizedString this[[NotNull] string key, params object[] arguments] =>
+            _localizer[key, arguments];
+
+        /// <summary>
+        /// Creates a new <see cref="IStringLocalizer"/> for a specific <see cref="CultureInfo"/>.
+        /// </summary>
+        /// <param name="culture">The <see cref="CultureInfo"/> to use.</param>
+        /// <returns>A culture-specific <see cref="IStringLocalizer"/>.</returns>
+        IStringLocalizer IStringLocalizer.WithCulture([NotNull] CultureInfo culture) => WithCulture(culture);
+
+        /// <inheritdoc />
+        public virtual LocalizedString GetString([NotNull] string key) => _localizer.GetString(key);
+
+        /// <inheritdoc />
+        public virtual LocalizedString GetString([NotNull] string key, params object[] arguments) =>
+            _localizer.GetString(key, arguments);
+
+        /// <inheritdoc />
+        public IEnumerable<LocalizedString> GetAllStrings(bool includeAncestorCultures) =>
+            _localizer.GetAllStrings(includeAncestorCultures);
+
+        /// <inheritdoc />
+        public virtual LocalizedHtmlString Html([NotNull] string key) => ToHtmlString(_localizer.GetString(key));
+
+        /// <inheritdoc />
+        public virtual LocalizedHtmlString Html([NotNull] string key, params object[] arguments)
+        {
+            var stringValue = _localizer[key].Value;
+
+            return ToHtmlString(new LocalizedString(key, EncodeArguments(stringValue, arguments)));
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="IHtmlLocalizer"/> for a specific <see cref="CultureInfo"/>.
+        /// </summary>
+        /// <param name="culture">The <see cref="CultureInfo"/> to use.</param>
+        /// <returns>A culture-specific <see cref="IHtmlLocalizer"/>.</returns>
+        public IHtmlLocalizer WithCulture([NotNull] CultureInfo culture) =>
+            new HtmlLocalizer(_localizer.WithCulture(culture), _encoder);
+
+        /// <summary>
+        /// Creates a new <see cref="LocalizedHtmlString"/> for a <see cref="LocalizedString"/>.
+        /// </summary>
+        /// <param name="result">The <see cref="LocalizedString"/>.</param>
+        protected LocalizedHtmlString ToHtmlString(LocalizedString result) =>
```

Comment:
This class seems to have a mix of protected/public methods, some of which are virtual. Is there a particular rule we're following to decide which is which? E.g. it's not super common in general to have protected-but-not-virtual methods (it's not by any means wrong - it's just not super common).